ocpi-toolkit-2.2.1/src/main/kotlin/com/izivia/ocpi/toolkit/modules/cdr/CdrsEmspServer.kt

```diff
@@ -0,0 +1,48 @@
+package com.izivia.ocpi.toolkit.modules.cdr
+
+import com.izivia.ocpi.toolkit.common.OcpiSelfRegisteringModuleServer
+import com.izivia.ocpi.toolkit.common.URL
+import com.izivia.ocpi.toolkit.common.httpResponse
+import com.izivia.ocpi.toolkit.common.mapper
+import com.izivia.ocpi.toolkit.modules.cdr.domain.Cdr
+import com.izivia.ocpi.toolkit.modules.versions.domain.InterfaceRole
+import com.izivia.ocpi.toolkit.modules.versions.domain.ModuleID
+import com.izivia.ocpi.toolkit.modules.versions.domain.VersionNumber
+import com.izivia.ocpi.toolkit.modules.versions.repositories.MutableVersionsRepository
+import com.izivia.ocpi.toolkit.transport.TransportServer
+import com.izivia.ocpi.toolkit.transport.domain.HttpMethod
+
+class CdrsEmspServer(
+    private val service: CdrsEmspInterface,
+    versionsRepository: MutableVersionsRepository? = null,
+    basePathOverride: String? = null
+) : OcpiSelfRegisteringModuleServer(
+    ocpiVersion = VersionNumber.V2_2_1,
+    moduleID = ModuleID.cdrs,
+    interfaceRole = InterfaceRole.RECEIVER,
+    versionsRepository = versionsRepository,
+    basePathOverride = basePathOverride
+) {
+    override suspend fun doRegisterOn(transportServer: TransportServer) {
+        transportServer.handle(
+            method = HttpMethod.GET,
+            path = basePathSegments
+        ) { req ->
+            req.httpResponse {
+                service.getCdr(req.baseUrl as URL)
+            }
+        }
```

Comment:
It's up to the user of the lib to implement their own endpoint. You can remove this. The user will have to either extend `CdrsEmspServer` and add their own endpoint or register their endpoint elsewhere